pkg/controller/datavolume/clone-controller.go

```diff
@@ -503,11 +524,32 @@ func (r *CloneReconciler) reconcileClone(log logr.Logger,
 		fallthrough
 	case SmartClone:
 		if !shouldBeMarkedWaitForFirstConsumer {
-			return r.finishClone(log, datavolume, pvc, pvcSpec, transferName, selectedCloneStrategy)
+			res, err := r.finishClone(log, &syncRes, transferName)
+			syncRes.result = &res
+			return syncRes, err
+
 		}
 	}
 
-	return r.reconcileDataVolumeStatus(datavolume, pvc, setCloneType, r.updateStatusPhase)
+	return syncRes, syncErr
+}
+
+func (r CloneReconciler) updateStatus(syncRes dataVolumeCloneSyncResult, syncErr error) (reconcile.Result, error) {
+	modifier := getModifier(syncRes)
+	ps := syncRes.phaseSync
+	if ps != nil {
+		syncErr = r.updateDataVolumeStatusPhaseWithEvent(ps.phase, syncRes.dv, ps.pvc, modifier, ps.event)
+	}
+	if syncErr != nil {
+		return reconcile.Result{}, syncErr
```

Comment:
Why return without attempting to check/update status?

Reply:
fixed